xchange-kraken/src/main/java/org/knowm/xchange/kraken/service/KrakenAccountService.java

```diff
@@ -92,50 +90,108 @@ public String withdrawFunds(WithdrawFundsParams params) throws IOException {
+      depositAddresses = getDepositAddresses(currency.toString(), "REP", newAddress);
     } else if (Currency.USD.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "SynapsePay (US Wire)", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "SynapsePay (US Wire)", newAddress);
     } else if (Currency.XDG.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "Dogecoin", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "Dogecoin", newAddress);
     } else if (Currency.MLN.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "MLN", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "MLN", newAddress);
     } else if (Currency.GNO.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "GNO", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "GNO", newAddress);
     } else if (Currency.QTUM.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "QTUM", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "QTUM", newAddress);
     } else if (Currency.XTZ.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "XTZ", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "XTZ", newAddress);
     } else if (Currency.ATOM.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "Cosmos", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "Cosmos", newAddress);
     } else if (Currency.EOS.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "EOS", false);
+      depositAddresses = getDepositAddresses(currency.toString(), "EOS", newAddress);
     } else if (Currency.DASH.equals(currency)) {
-      depositAddresses = getDepositAddresses(currency.toString(), "Dash", false);
-    } else {
-      throw new RuntimeException("Not implemented yet, Kraken works only for BTC and LTC");
+      depositAddresses = getDepositAddresses(currency.toString(), "Dash", newAddress);
+    }
+
+    return depositAddresses;
+  }
+
+  KrakenDepositAddress[] getDepositAddresses(Currency currency, String ...args) throws IOException {
+    String protocol = null;
+
+    if(args != null && args.length > 0) {
+      protocol = args[0];
+    }
+
+    Currency krakenCurrency = Currency.getInstance(KrakenUtils.getKrakenCurrencyCode(currency));
+
+    KrakenDepositAddress[] depositAddresses = requestDepositAddress(false, krakenCurrency, protocol);
+    if(depositAddresses == null || depositAddresses.length <= 0) {
+      depositAddresses = requestDepositAddress(true, krakenCurrency, protocol);
+    }
+
+    if(depositAddresses == null || depositAddresses.length <= 0) {
+      throw new IOException("Unable to create wallet currency: "+currency+" protocol: "+protocol);
```

Comment:
this and many other places have wrong formatting 

build your code using maven - it will automaticaly fix it to our guidelines